components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -329,25 +347,84 @@ private void addEndpoint(ActionDTO actionDTO, Integer tenantId) throws ActionMgt
     private void updateEndpoint(ActionDTO updatingActionDTO, ActionDTO existingActionDTO, Integer tenantId)
             throws ActionMgtException {
 
-        EndpointConfig updatingEndpoint = updatingActionDTO.getEndpoint();
-        if (updatingEndpoint == null) {
+        if (updatingActionDTO.getEndpoint() == null) {
             return;
         }
 
         try {
-            if (updatingEndpoint.getUri() != null) {
-                updateActionPropertiesInDB(updatingActionDTO.getId(),
-                        Collections.singletonMap(URI_PROPERTY, new ActionProperty.BuilderForDAO(
-                                updatingEndpoint.getUri()).build()), tenantId);
-            }
+            // Deletes existing endpoint config.
+            deleteEndpoint(existingActionDTO.getId(), existingActionDTO.getEndpoint(), tenantId);
+
+            ActionDTO actionDTO = actionMgtDAOUtil.buildActionDTOWithEndpoint(updatingActionDTO, existingActionDTO);
+            EndpointConfig updatingEndpoint = actionDTO.getEndpoint();
+
+            Map<String, ActionProperty> updatingEndpointProperties = new HashMap<>();
+            try {
+                updatingEndpointProperties.put(URI_PROPERTY,
+                        new ActionProperty.BuilderForDAO(updatingEndpoint.getUri()).build());
+                updatingEndpointProperties.put(AUTHN_TYPE_PROPERTY,
+                        new ActionProperty.BuilderForDAO(
+                                updatingEndpoint.getAuthentication().getType().name()).build());
+                updatingEndpoint.getAuthentication().getProperties().forEach(authProperty ->
+                        updatingEndpointProperties.put(authProperty.getName(),
+                                new ActionProperty.BuilderForDAO(authProperty.getValue()).build()));
+
+                // Allowed headers and parameters are optional properties.
+                if (updatingEndpoint.getAllowedHeaders() != null) {
+                    updatingEndpointProperties.put(ALLOWED_HEADERS_PROPERTY,
+                            actionMgtDAOUtil.buildActionPropertyFromList(updatingEndpoint.getAllowedHeaders()));
+                }
+                if (updatingEndpoint.getAllowedParameters() != null) {
+                    updatingEndpointProperties.put(ALLOWED_PARAMETERS_PROPERTY,
+                            actionMgtDAOUtil.buildActionPropertyFromList(updatingEndpoint.getAllowedParameters()));
+                }
```

Comment:
Let's move this logic to the above method in `ActionMgtDAOUtil` class.
And lets rename that method name to `getUpdatingEndpointProperties`